### PR TITLE
Support Terrain in Globe projection

### LINKED ISSUE #4927: [Feat] Support Terrain in Globe projection
There's a lack of Terrain3D Support on the GLobe:

https://github.com/user-attachments/assets/365b88ab-de47-441a-981f-c6d756ccc38a

I'm working on this working as expected:

https://github.com/user-attachments/assets/aea39205-aff6-491b-81eb-f0ff649e07b0

**These layers are working now**
- circle
- line - #4970 
- raster
- symbol

There work underway. After that PR these layers will work to:
- #4977 

- background 
- fill
- fill-extrusion
- heatmap
- hillshade
- custom (partially)

Other changes that are impacting this feature:
- #4779 
- #4937 
- #4924 
- #4868 
- #4963 / #4928

### LINKED ISSUE #4928: The unsupported Fog/Horizon is isn't disabled properly on Globe with Terrain3D
Repro here:
- #4926 

Changing the pitch will remove all texture, resulting in a white screen:

https://github.com/user-attachments/assets/25a589f3-a148-4a23-a108-84f977ab0a47

### PR DESCRIPTION
**Pull request by birkskyum**
Closes #4927 
- #4927 

Closes #4928 
- #4928 

After

https://github.com/user-attachments/assets/151af953-f7ea-4b9d-8f5a-82bea15a9ca0

Before

https://github.com/user-attachments/assets/61fcd41c-e91c-48bf-909a-922d29402c66

## Launch Checklist

 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [X] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/render/draw_heatmap.ts [hunk 1]
```diff
@@ -1,23 +1,23 @@
-import { Texture } from './texture';
-import { Color } from '@maplibre/maplibre-gl-style-spec';
-import { DepthMode } from '../gl/depth_mode';
-import { StencilMode } from '../gl/stencil_mode';
-import { ColorMode } from '../gl/color_mode';
-import { CullFaceMode } from '../gl/cull_face_mode';
-import { Context } from '../gl/context';
-import { Framebuffer } from '../gl/framebuffer';
-import { Tile } from '../source/tile';
+import {Texture} from './texture';
```
Comment (HarelM) [inline src/render/draw_heatmap.ts:1]: This import changes creates noise in this PR, can you fix this as part of a different PR, hopefully using a lint rule?

**Comment by birkskyum** [comment]
@HarelM , I'm having some render tests for the  fill layer fail in CI-only, with very small diffs on Ubuntu / Windows.

I believe it's just platform differences. How can I get render output from the CI so they can be added as i.e. expected-ubuntu and expected-windows?

**Comment by birkskyum** [comment]
### Review notes

**Render tests**
I've copied the folder `/projection/globe/` -> `/projection/globe/terrain/`, and added terrain and a terrain source in all of them. Apart from that they're the same.

**Changes to terrain**
These changes are to resolve the issue with the fog not being supported on Globe Mode. The previous solution made the texture color transparent, which means that not the mountains are transparent - A better solution, which is applied here, is to pass the "globe mode" flag ot the terrain shader so that the fog logic isn't executed.

File: src/render/program/terrain_program.ts [hunk 2]
```diff
@@ -85,7 +87,8 @@ const terrainUniformValues = (
     // Set opacity to 0 when in globe mode to disable fog
     'u_fog_ground_blend_opacity': isGlobeMode ? 0 : (sky ? sky.calculateFogBlendOpacity(pitch) : 0),
     'u_horizon_color': sky ? sky.properties.get('horizon-color') : Color.white,
-    'u_horizon_fog_blend': sky ? sky.properties.get('horizon-fog-blend') : 1
+    'u_horizon_fog_blend': sky ? sky.properties.get('horizon-fog-blend') : 1,
+    'u_is_globe_mode': isGlobeMode ? 1 : 0
```
Comment (HarelM) [inline src/render/program/terrain_program.ts:91]: Is the change for `u_fog_ground_blend_opacity` needed if we pass that as a boolean now?

**Review by HarelM — APPROVED** [review]
Looks great, nice work!
I added a minor comment.

File: src/render/program/terrain_program.ts (hunk 2 above)
Comment (birkskyum) [reply src/render/program/terrain_program.ts:91]: yes, it's still needed, otherwise the pitch issue appear again.

**Pull request merged**